### PR TITLE
Improve readability of timekeeping code in idle server check

### PR DESCRIPTION
**Pull request by DomGarguilo**
Tried to improve things by refactoring the code to make things more readable. Also converted the longs to use the `NanoTime` object.

The logic should be identical before and after this change.

I did notice that the compactor idle check seems to include the same check that is happening in the code that reads it (`AbstractServer.idleProcessCheck()`) where it checks the `idleReportingPeriod` has elapsed since the last compaction has completed before returning true. Unless there is a reason this was added, it seems like that part could just be removed and the same check would take place upstream.

### CODE REVIEW

File: server/base/src/main/java/org/apache/accumulo/server/AbstractServer.java [hunk 1]
```diff
@@ -71,28 +73,28 @@ protected AbstractServer(String appName, ConfigOpts opts, String[] args) {
         lmd.getIntervalMillis(context.getConfiguration()), TimeUnit.MILLISECONDS);
     ThreadPools.watchNonCriticalScheduledTask(future);
     processMetrics = new ProcessMetrics(context);
-    idleReportingPeriodNanos = TimeUnit.MILLISECONDS.toNanos(
+    idleReportingPeriod = Duration.ofMillis(
```
Comment (keith-turner) [inline server/base/src/main/java/org/apache/accumulo/server/AbstractServer.java:76]: Not a change for this PR, but looking at this I suspect adding a `Duration getDuration(Property)` to Accumulo configuration would improve this and other code.

File: server/compactor/src/main/java/org/apache/accumulo/compactor/Compactor.java [hunk 2]
```diff
@@ -694,15 +693,17 @@ public void run() {
...
 
       final AtomicReference<Throwable> err = new AtomicReference<>();
-      final AtomicLong timeSinceLastCompletion = new AtomicLong(0L);
+      final AtomicReference<NanoTime> lastCompletionTime = new AtomicReference<>(null);
 
       while (!shutdown) {
 
         idleProcessCheck(() -> {
-          return timeSinceLastCompletion.get() == 0
-              /* Never started a compaction */ || (timeSinceLastCompletion.get() > 0
```
Comment (keith-turner) [inline server/compactor/src/main/java/org/apache/accumulo/compactor/Compactor.java:703]: These changes fix a bug in the existing code. Seems like the existing code was assuming that nano times would be positive which I do not think is safe to assume.   This change is being made in main, if the bug exists in 2.1 then may want to fix it in another PR w/o the NanoTime changes.

**Comment by DomGarguilo** [comment]
Converting to draft for now. Going to make another PR in 2.1 to clean up the duplicate logic outline above and merge that up first.

**Review by keith-turner — APPROVED** [review]

**Review by ctubbsii — APPROVED** [review]
Looks fine to me

**Pull request merged**